File: Components/Dashboard/VB/ASPxDashboard/App_Code/Data/DataLoader.vb
Imports Microsoft.VisualBasic
Imports System.Collections.Generic
Imports System.Data
Imports System
Imports DevExpress.Utils
Imports System.IO
Imports System.Web.Hosting
Imports System.Configuration
Imports DevExpress.DashboardWeb

Namespace DashboardMainDemo
	Public NotInheritable Class DataLoader
		Private Sub New()
		End Sub
		Private Shared Function GetRelativePath(ByVal name As String) As String
			Return Path.Combine(HostingEnvironment.MapPath("~"), "App_Data", "Data", name)
		End Function
		Private Shared Function LoadData(ByVal fileName As String) As DataSet
			Dim path As String = GetRelativePath(String.Format("{0}.xml", fileName))
			Dim ds As New DataSet()
			ds.ReadXml(path, XmlReadMode.ReadSchema)
			Return ds
		End Function
		Public Shared Function LoadSales() As DataSet
			Return LoadData("DashboardSales")
		End Function
		Public Shared Function LoadEmployees() As DataSet
			Return LoadData("DashboardEmployeesAndDepartments")
		End Function
		Public Shared Function LoadCustomerSupport() As DataSet
			Return LoadData("DashboardCustomerSupport")
		End Function
		Public Shared Function LoadRevenueByIndustry() As DataSet
			Return LoadData("DashboardRevenueByIndustry")
		End Function
		Public Shared Function LoadEuroEnergyStatistics() As DataSet
			Return LoadData("DashboardEnergyStatictics")
		End Function
		Public Shared Function LoadEnergyConsumptionTotal() As DataSet
			Return LoadData("DashboardEnergyConsumptionTotal")
		End Function
		Public Shared Function LoadEnergyConsumptionBySector() As DataSet
			Return LoadData("DashboardEnergyConsumptionBySector")
		End Function
		Public Shared Function LoadChampionsLeagueStatistics() As DataSet
			Return LoadData("DashboardChampionsLeagueStatistics")
		End Function
		Public Shared Function GetProductDetailsXmlDataSource() As String
			Return GetRelativePath(String.Format("{0}.xml", "DashboardProductDetails"))
		End Function
		Public Shared Function GetImagesFolder() As String
			Return GetRelativePath("ProductDetailsImages")
		End Function

		Public Shared Sub LoadData(ByVal e As DataLoadingWebEventArgs)
			Select Case e.DashboardId
				Case "WebsiteStatistics"
					Dim data As New WebsiteStatisticsDataGenerator()
					e.Data = data.WebsiteStatistics
				Case "SalesPerformance"
					Dim dataGenerator As New SalesPerformanceDataGenerator(LoadSales())
					dataGenerator.Generate()
					If e.DataSourceComponentName = "dsMonthlySales" Then
						e.Data = dataGenerator.MonthlySales
					ElseIf e.DataSourceComponentName = "dsTotalSales" Then
						e.Data = dataGenerator.TotalSales
					ElseIf e.DataSourceComponentName = "dsKeyMetrics" Then
						e.Data = dataGenerator.KeyMetrics
					End If
				Case "HumanResources"
					Dim humanResourcesData As New HumanResourcesData(LoadEmployees())
					If e.DataSourceComponentName = "dsEmployees" Then
						e.Data = humanResourcesData.EmployeeData
					ElseIf e.DataSourceComponentName = "dsDepartments" Then
						e.Data = humanResourcesData.DepartmentData
					End If
				Case "EnergyStatistics"
					e.Data = EnergyStaticticsDataHelper.Generate(LoadEuroEnergyStatistics())
				Case "SalesOverview"
					Dim salesOverviewDataGenerator As New SalesOverviewDataGenerator(LoadSales())
					salesOverviewDataGenerator.Generate()
					e.Data = salesOverviewDataGenerator.Data
				Case "SalesDetails"
					Dim salesDetailsDataGenerator As New SalesDetailsDataGenerator(LoadSales())
					salesDetailsDataGenerator.Generate()
					e.Data = salesDetailsDataGenerator.Data
				Case "RevenueByIndustry"
					e.Data = RevenueByIndustryDataHelper.Generate(LoadRevenueByIndustry())
				Case "ChampionsLeagueStatistics"
					e.Data = ChampionsLeagueStatisticsDataHelper.Generate(LoadChampionsLeagueStatistics())
				Case "RevenueAnalysis"
					Dim revenueAnalysisDataGenerator As New RevenueAnalysisDataGenerator(LoadSales())
					revenueAnalysisDataGenerator.Generate()
					e.Data = revenueAnalysisDataGenerator.Data
				Case "CustomerSupport"
					Dim customerSupportData As New CustomerSupportData(LoadCustomerSupport(), LoadEmployees())
					e.Data = customerSupportData.CustomerSupport
				Case "EnergyConsumption"
					If e.DataSourceComponentName = "dsConsumptionTotal" Then
						e.Data = EnergyConsumptionDataHelper.GenerateTotal(LoadEnergyConsumptionTotal())
					ElseIf e.DataSourceComponentName = "dsConsumptionBySector" Then
						e.Data = EnergyConsumptionDataHelper.GenerateBySector(LoadEnergyConsumptionBySector())
					End If
			End Select
		End Sub
	End Class
End Namespace

File: Components/Dashboard/CS/ASPxDashboard/App_Code/Data/DataLoader.cs
using System.Collections.Generic;
using System.Data;
using System;
using DevExpress.Utils;
using System.IO;
using System.Web.Hosting;
using System.Configuration;
using DevExpress.DashboardWeb;

namespace DashboardMainDemo {
    public static class DataLoader {
        static string GetRelativePath(string name) {
            return Path.Combine(HostingEnvironment.MapPath("~"), "App_Data", "Data", name);
        }
        static DataSet LoadData(string fileName) {
            string path = GetRelativePath(string.Format("{0}.xml", fileName));
            DataSet ds = new DataSet();
            ds.ReadXml(path, XmlReadMode.ReadSchema);
            return ds;
        }
        public static DataSet LoadSales() {
            return LoadData("DashboardSales");
        }
        public static DataSet LoadEmployees() {
            return LoadData("DashboardEmployeesAndDepartments");
        }
        public static DataSet LoadCustomerSupport() {
            return LoadData("DashboardCustomerSupport");
        }
        public static DataSet LoadRevenueByIndustry() {
            return LoadData("DashboardRevenueByIndustry");
        }
        public static DataSet LoadEuroEnergyStatistics() {
            return LoadData("DashboardEnergyStatictics");
        }
        public static DataSet LoadEnergyConsumptionTotal() {
            return LoadData("DashboardEnergyConsumptionTotal");
        }
        public static DataSet LoadEnergyConsumptionBySector() {
            return LoadData("DashboardEnergyConsumptionBySector");
        }
        public static DataSet LoadChampionsLeagueStatistics() {
            return LoadData("DashboardChampionsLeagueStatistics");
        }
        public static string GetProductDetailsXmlDataSource() {
            return GetRelativePath(String.Format("{0}.xml", "DashboardProductDetails"));
        }
        public static string GetImagesFolder() {
            return GetRelativePath("ProductDetailsImages");
        }

        public static void LoadData(DataLoadingWebEventArgs e) {
            switch(e.DashboardId) {
                case "WebsiteStatistics":
                    WebsiteStatisticsDataGenerator data = new WebsiteStatisticsDataGenerator();
                    e.Data = data.WebsiteStatistics;
                    break;
                case "SalesPerformance":
                    SalesPerformanceDataGenerator dataGenerator = new SalesPerformanceDataGenerator(LoadSales());
                    dataGenerator.Generate();
                    if(e.DataSourceComponentName == "dsMonthlySales")
                        e.Data = dataGenerator.MonthlySales;
                    else if(e.DataSourceComponentName == "dsTotalSales")
                        e.Data = dataGenerator.TotalSales;
                    else if(e.DataSourceComponentName == "dsKeyMetrics")
                        e.Data = dataGenerator.KeyMetrics;
                    break;
                case "HumanResources":
                    HumanResourcesData humanResourcesData = new HumanResourcesData(LoadEmployees());
                    if(e.DataSourceComponentName == "dsEmployees")
                        e.Data = humanResourcesData.EmployeeData;
                    else if(e.DataSourceComponentName == "dsDepartments")
                        e.Data = humanResourcesData.DepartmentData;
                    break;
                case "EnergyStatistics":
                    e.Data = EnergyStaticticsDataHelper.Generate(LoadEuroEnergyStatistics());
                    break;
                case "SalesOverview":
                    SalesOverviewDataGenerator salesOverviewDataGenerator = new SalesOverviewDataGenerator(LoadSales());
                    salesOverviewDataGenerator.Generate();
                    e.Data = salesOverviewDataGenerator.Data;
                    break;
                case "SalesDetails":
                    SalesDetailsDataGenerator salesDetailsDataGenerator = new SalesDetailsDataGenerator(LoadSales());
                    salesDetailsDataGenerator.Generate();
                    e.Data = salesDetailsDataGenerator.Data;
                    break;
                case "RevenueByIndustry":
                    e.Data = RevenueByIndustryDataHelper.Generate(LoadRevenueByIndustry());
                    break;
                case "ChampionsLeagueStatistics":
                    e.Data = ChampionsLeagueStatisticsDataHelper.Generate(LoadChampionsLeagueStatistics());
                    break;
                case "RevenueAnalysis":
                    RevenueAnalysisDataGenerator revenueAnalysisDataGenerator = new RevenueAnalysisDataGenerator(LoadSales());
                    revenueAnalysisDataGenerator.Generate();
                    e.Data = revenueAnalysisDataGenerator.Data;
                    break;
                case "CustomerSupport":
                    CustomerSupportData customerSupportData = new CustomerSupportData(LoadCustomerSupport(), LoadEmployees());
                    e.Data = customerSupportData.CustomerSupport;
                    break;
                case "EnergyConsumption":
                    if(e.DataSourceComponentName == "dsConsumptionTotal")
                        e.Data = EnergyConsumptionDataHelper.GenerateTotal(LoadEnergyConsumptionTotal());
                    else if(e.DataSourceComponentName == "dsConsumptionBySector")
                        e.Data = EnergyConsumptionDataHelper.GenerateBySector(LoadEnergyConsumptionBySector());
                    break;
            }
        }
    }
}
